### PR TITLE
librbd: fix Clang warnings

### PR DESCRIPTION
**Pull request by wjwithagen**
- Double declaration
- Unneeded std::move()
```
/home/jenkins/workspace/ceph-master-compile/src/testk_HttpClient.cc:34:
/home/jenkins/workspace/ceph-master-compile/src/librbd/migration/HttpClienng a temporary object prevents copy elision [-Wpessimizing-move]
    response = std::move(m_parser->release());
               ^
/home/jenkins/workspace/ceph-master-compile/src/librbd/migration/HttpClienstd::move call here
    response = std::move(m_parser->release());
```

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by tchaikov — APPROVED** [review]

**Comment by wjwithagen** [comment]
@tchaikov 
I guess we need to ask @adamemerson as well, since he suggested that copy was not needed 
for the changes in #38308.
If that is the case, we need to find a way to sillence Clang here.

**Comment by tchaikov** [comment]
> @tchaikov
> I guess we need to ask @adamemerson as well, since he suggested that copy was not needed
> for the changes in #38308.
> If that is the case, we need to find a way to sillence Clang here.

i think you misunderstood his comment. his comment does not apply to the change in this PR. but anyway...

**Review by adamemerson — APPROVED** [review]
Looks good to me

File: src/librbd/migration/HttpClient.cc [hunk 1]
```diff
@@ -460,7 +460,7 @@ class HttpClient<I>::HttpSession : public HttpSessionInterface {
     if (work->header_only()) {
       m_parser.emplace(std::move(*m_header_parser));
     }
-    response = std::move(m_parser->release());
+    response = m_parser->release();
```
Comment (dillaman) [inline src/librbd/migration/HttpClient.cc:463]: Just for my own education, what was the clang warning here? I was under the (mistaken?) impression that copy elision didn't function for assignment.

Comment (dillaman) [reply src/librbd/migration/HttpClient.cc:463]: Disregard -- you put it in the commit message.

**Review by dillaman — APPROVED** [review]

**Pull request merged**